### PR TITLE
[WIP] add dependency check maven plugin

### PR DESCRIPTION
**Pull request by ZheSun88**
related to #1349 

This change is [Reviewable](https://reviewable.io/reviews/vaadin/platform/1350)

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -145,6 +145,26 @@
                 </plugin>
             </plugins>
         </pluginManagement>
-    </build>
 
+        <plugins>
```
Comment (manolo) [inline pom.xml:149]: Apparently it's gonna be executed in all builds, I wouldn't run this always but in CI before beta or rc releases it is enough to run the goal `org.owasp:dependency-check-maven:5.2.2:check` from command line

Comment (ZheSun88) [reply pom.xml:149]: Currently, this is only a test for collecting ideas. 
The plan is adding this as a profile, then do the check for the necessary steps. 
running this before beta or rc release is fine, as long as they are not too late. I'd prefer to move the checking even earlier. 

also, I am planning to make use of 
```
<configuration>
        <failBuildOnCVSS>8</failBuildOnCVSS>
</configuration>
```
to limit the vulnerabilities, as there might be some false positive reports.

Comment (TatuLund) [reply pom.xml:149]: Just for information, it is also possible to exclude test and provided scope dependencies from the check if that is desired.

Comment (ZheSun88) [reply pom.xml:149]: oh, i will check the result after excluding the test and provided dependencies.

Comment (ZheSun88) [reply pom.xml:149]: difference between skipping the test/provided dependencies is trivial. 
one solution here can be that we pin those dependencies in platform level. 
introduce a `<dependencyManagement>` in the root, then declare those in submodules. 

a proper way would be fix those from components

**Comment by ZheSun88** [comment]
Finally [this issue](https://github.com/jeremylong/DependencyCheck/issues/2651) has been fixed. so we need to update plugin version to 6.0.3 and test

**Pull request closed without merging**